apps/sms/views/conversation/js/conversation.js

```diff
@@ -815,8 +816,9 @@ var ConversationView = {
     // Update the stored thread data
     Threads.registerMessage(message);
 
-    this.appendMessage(message);
-    TimeHeaders.updateAll('header[data-time-update]');
+    return this.appendMessage(message).then(
+      () => TimeHeaders.updateAll('header[data-time-update]')
```

Comment:
hmm... maybe it's for updating the headers in inbox view. Not sure we really need the updateAll here either but we definitely need to revisit this time update mechanism once we encapsulate the view separately.